### PR TITLE
fix(hydration): do not warn for escaped `<style>` content

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

Fixes the case where the contents of a `<style>` tag are HTML-escaped when serialized on the server, leading to a hydration mismatch warning on the client, where it is not escaped.

The serialization logic uses `html-escape.ts`, so I based my solution on the same thing:

https://github.com/salesforce/lwc/blob/0666b1613c3124837a9acb79f1fc6eee538015c6/packages/%40lwc/shared/src/html-escape.ts#L8-L20

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

## GUS work item
W-12477791

### CODE REVIEW

**Comment by nolanlawson** [comment]
Rather than putting the new logic in the hydration mismatch check, I considered putting it directly in `stylesheet.ts` where the vnode is generated:

https://github.com/salesforce/lwc/blob/0666b1613c3124837a9acb79f1fc6eee538015c6/packages/%40lwc/engine-core/src/framework/stylesheet.ts#L40-L51

However, this doesn't seem to work. Escaping the `content` here just leads to a double-escape. So I figure it's easier to just change the hydration mismatch logic.

File: packages/@lwc/engine-core/src/framework/hydration.ts [hunk 1]
```diff
@@ -106,15 +106,49 @@ function hydrateNode(node: Node, vnode: VNode, renderer: RendererAPI): Node | nu
...
+function textNodeContentsAreEqual(node: Node, vnode: VText, renderer: RendererAPI): boolean {
+    const { getProperty } = renderer;
+    const nodeValue = getProperty(node, NODE_VALUE_PROP);
+
+    if (nodeValue === vnode.text) {
+        return true;
+    }
+
+    // Special case for empty text nodes – these are serialized differently on the server
+    if (nodeValue === '\u200D' && vnode.text === '') {
```
Comment (divmain) [inline packages/@lwc/engine-core/src/framework/hydration.ts:121]: I had to go back to the [original PR](https://github.com/salesforce/lwc/pull/2656) to remember why this is true. Thanks for adding context here.

**Review by divmain — APPROVED** [review]
LGTM. Thanks Nolan!

File: packages/@lwc/engine-core/src/framework/hydration.ts [hunk 2]
```diff
@@ -106,15 +106,49 @@ function hydrateNode(node: Node, vnode: VNode, renderer: RendererAPI): Node | nu
...
+
+function textNodeContentsAreEqual(node: Node, vnode: VText, renderer: RendererAPI): boolean {
+    const { getProperty } = renderer;
+    const nodeValue = getProperty(node, NODE_VALUE_PROP);
+
+    if (nodeValue === vnode.text) {
+        return true;
+    }
+
+    // Special case for empty text nodes – these are serialized differently on the server
```
Comment (nolanlawson) [inline packages/@lwc/engine-core/src/framework/hydration.ts:120]:
```suggestion
    // Special case for empty text nodes – these are serialized differently on the server
    // See https://github.com/salesforce/lwc/pull/2656
```

File: packages/@lwc/engine-core/src/framework/hydration.ts (hunk 1 above)
Comment (nolanlawson) [reply packages/@lwc/engine-core/src/framework/hydration.ts:122]: Thanks for the link! Added it to the comment.

**Pull request merged**